src/components/ImageView/index.tsx

```diff
@@ -1,36 +1,62 @@
+import purify from 'dompurify';
 import log from 'loglevel';
 import * as React from 'react';
 
 import styles from './styles.module.scss';
 
-type PublicProps = {
+export type PublicProps = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   mimeType: string;
   content: string;
 };
 
 type ConvertImageDataParams = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   content: string;
+  mimeType: string;
 };
 
-export const convertImageData = ({
+const convertImageData = ({
   _btoa = btoa,
   _log = log,
+  _sanitize = purify.sanitize,
   content,
+  mimeType,
 }: ConvertImageDataParams): string | null => {
+  const mimeTypeCased = mimeType.toLowerCase();
+  if (!mimeTypeCased.startsWith('image')) {
+    throw new Error(`Unexpected mimeType: "${mimeType}"`);
+  }
+
   try {
-    return _btoa(content);
+    return _btoa(
+      // Run sanitize on svg files.
+      mimeTypeCased === 'image/svg+xml' ? _sanitize(content) : content,
+    );
   } catch (error) {
-    _log.debug('Error converting image data to ascii:', error);
+    _log.error('Error converting image data to ascii:', error);
```

Comment:
> I did a quick test, and it looks like if we throw an error here, then the entire site is replaced with an error message. It looks like this is a development-only thing, but I'm not sure why the message isn't displayed inside the site. 

How would I reproduce this? I am trying out the patch locally and I can't get it to throw an error after editing the code and viewing http://localhost:3000/en-US/browse/494431/versions/1532144/

Reply:
Here's a patch I used to force an error:

```diff
diff --git a/src/components/ImageView/index.tsx b/src/components/ImageView/index.tsx
index fd37d13..a858781 100644
--- a/src/components/ImageView/index.tsx
+++ b/src/components/ImageView/index.tsx
@@ -21,7 +21,9 @@ type ConvertImageDataParams = {
 };
 
 const convertImageData = ({
-  _btoa = btoa,
+  _btoa = () => {
+    throw new Error('an error from btoa');
+  },
   _log = log,
   _sanitize = purify.sanitize,
   content,
@@ -38,8 +40,7 @@ const convertImageData = ({
       mimeTypeCased === 'image/svg+xml' ? _sanitize(content) : content,
     );
   } catch (error) {
-    _log.error('Error converting image data to ascii:', error);
-    return null;
+    throw new Error(`error: ${error.message}`);
   }
 };
```

Wnen I visit this URL: http://localhost:3000/en-US/browse/494431/versions/1532144/?path=skin%2Fabb-logo.png I see it as a full page error.